### PR TITLE
feature-8950: Make form fields editable after order

### LINKED ISSUE #8950: Make form fields editable after order
Currently some form fields are editable after an order, some fields are not editable.

1. Please ensure all form fields are potentially editable. Test on this event https://eventyay.com/e/bdaa2bc5
2. Add a column with the option "allow editing" into the attendee form builder behind "include" and "require".
3. Attendee, organizer, system admins should have permissions to edit the form elements.

Screenshot from 2023-07-10 20-46-20

Screenshot from 2023-07-10 20-39-52

### PR DESCRIPTION
**Pull request by lthanhhieu**
Fixes #8950 

#### Short description of what this resolves:
- Make form fields editable after order

#### Changes proposed in this pull request:
- Make form fields editable after order

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/components/forms/orders/order-form.hbs [hunk 1]
```diff
@@ -77,221 +77,22 @@
...
-                  />
-                {{/if}}
+                <Forms::Orders::EditableFields 
+                  @field={{field}}
+                  @fields={{this.allFields.attendee}}
+                  @data={{this.data}}
+                  @holder={{holder}}
+                  @index={{index}}
+                  @countries={{this.countries}} 
+                  @languageForms1={{languageForms1}}
```
Comment (cweitat) [inline app/templates/components/forms/orders/order-form.hbs:87]: hi pls use meaningful naming

**Review by cweitat — CHANGES_REQUESTED** [review]

File: app/components/forms/orders/attendee-list.js [hunk 2]
```diff
@@ -17,20 +23,24 @@ export default class AttendeeList extends Component {
     return this.data.user;
   }
 
+  @tracked languageForms1 = languageForms1;
```
[the remark this answers is not in the document]
Comment (khangon) [reply app/components/forms/orders/attendee-list.js]: Hi @cweitat,
Other names with full meanings have been used. We must preserve it for existing events to guarantee that updating does not disrupt the app.

[the remark this answers is not in the document]
Comment (lthanhhieu) [reply app/components/forms/orders/attendee-list.js]: We fixed it.

**Comment by mariobehling** [comment]
Please also resolve conflicts.

File: app/components/forms/orders/attendee-list.js [hunk 3]
```diff
@@ -17,20 +23,24 @@ export default class AttendeeList extends Component {
...
-        this.languageFormMapCodeToName(attendee, 'native_language', languageForms1);
+        this.languageFormMapCodeToName(attendee, 'native_language', this.nativeLanguage);
       }
       if (attendee.fluent_language) {
-        this.languageFormMapCodeToName(attendee, 'fluent_language', languageForms2);
+        this.languageFormMapCodeToName(attendee, 'fluent_language', this.fluentLanguage);
       }
       if (attendee.language_form_1) {
-        this.languageFormMapCodeToName(attendee, 'language_form_1', languageForms1);
+        this.languageFormMapCodeToName(attendee, 'language_form_1', this.nativeLanguage);
```
Comment (cweitat) [inline app/components/forms/orders/attendee-list.js:40]: what's with language_form_1 and language_form_2?

**Review by cweitat — CHANGES_REQUESTED** [review]
rest of the code LGTM

File: app/components/forms/orders/attendee-list.js (hunk 3 above)
Comment (khangon) [reply app/components/forms/orders/attendee-list.js:40]: https://github.com/fossasia/open-event-frontend/issues/8866

For this ticket. We change the fieldIdentifier from language_form_1 to the new fieldIdentifier native_language.
However, some data was saved with language_form_1. So we also need to accommodate them

Comment (khangon) [reply app/components/forms/orders/attendee-list.js:40]: https://github.com/fossasia/open-event-frontend/issues/8866

For this ticket. We change the fieldIdentifier from language_form_1 to the new fieldIdentifier native_language.
However, some data was saved with language_form_1. So we also need to accommodate them

**Review by cweitat — APPROVED** [review]

**Comment by cweitat** [comment]
@mariobehling pls test. else code looks good to me

**Pull request merged**